prime-router/src/main/kotlin/azure/MessagesFunction.kt

```diff
@@ -0,0 +1,125 @@
+import gov.cdc.prime.router.Message
+import gov.cdc.prime.router.common.JacksonMapperUtilities
+import gov.cdc.prime.router.tokens.AuthenticatedClaims
+import gov.cdc.prime.router.tokens.authenticationFailure
+import gov.cdc.prime.router.tokens.authorizationFailure
+import org.apache.logging.log4j.kotlin.Logging
+
+const val MESSAGE_SEARCH_ID_PARAMETER = "messageId"
+
+/**
+ * Azure Functions with HTTP Trigger.
+ * Search and retrieve messages
+ */
+class MessagesFunction(
+    private val dbAccess: DatabaseAccess = DatabaseAccess(),
+) : Logging {
+
+    /**
+     * entry point for the /messages/search endpoint,
+     * which searches for a given messageId in the database
+     */
+    @FunctionName("messageSearch")
+    @StorageAccount("AzureWebJobsStorage")
+    fun run(
+        @HttpTrigger(
+            name = "messageSearch",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "messages/search"
+        ) request: HttpRequestMessage<String?>
+    ): HttpResponseMessage {
+
+        return try {
+            val claims = AuthenticatedClaims.authenticate(request)
+                ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+
+            // admin permissions only
+            if (!claims.isPrimeAdmin) {
+                return HttpUtilities.unauthorizedResponse(request, authorizationFailure)
+            }
+
+            processSearchRequest(request)
+        } catch (ex: Exception) {
+            if (ex.message != null) {
+                logger.error(ex.message!!, ex)
+            } else {
+                logger.error(ex)
+            }
+            HttpUtilities.internalErrorResponse(request)
+        }
+    }
+
+    /**
+     * Handles an incoming message search request.
+     * @param request The incoming request
+     * @return Returns an HttpResponseMessage indicating the result of the operation and any resulting information
+     */
+    internal fun processSearchRequest(
+        request: HttpRequestMessage<String?>
+    ): HttpResponseMessage {
+
+        val messageId = request.queryParameters
+            .getOrDefault(
+                MESSAGE_SEARCH_ID_PARAMETER,
+                null
+            )
+
+        // return a list of messages or an error
+        var response: Any = emptyList<Message>()
+        var errorMessage: String? = ""
+
+        val httpStatus: HttpStatus =
+            try {
+                if (messageId.isNullOrEmpty() || messageId.isBlank()) {
+                    errorMessage = InvalidParamMessage("Missing the messageId request param").message
+                    HttpStatus.BAD_REQUEST
+                } else {
+                    val results = dbAccess.fetchCovidResultMetadatasByMessageId(messageId)
+                    response = results
+                        .map {
+                            Message(
+                                it.messageId,
+                                it.senderId,
```

Comment:
Will this return the Sender name? or just the senderId?

Reply:
sender id and client, which is in the mock ups 
![image](https://user-images.githubusercontent.com/19178435/196298577-ddf1fc40-cecc-4fad-bc04-275b6827b9ca.png)